### PR TITLE
Fix try haskell on Android

### PR DESCRIPTION
**Pull request by andys8**
## Fix

Cherrypicked fix provided in https://github.com/chrisdone/jquery-console/pull/71.

Tested on desktop, iOS, and Android - with Android now working. There is also another [comment](https://github.com/chrisdone/jquery-console/pull/71#issuecomment-364102640) confirming the fix.

Fixes https://github.com/haskell-infra/www.haskell.org/issues/252

## Review

I would welcome another person making sure "try it" is still working on desktop as before (comparing with currently deployed https://haskell.org).

Hint for review: I needed to change the listening host to `0.0.0.0` in order to be able to access the local dev server via my mobile device in the same network.

```patch
diff --git a/builder/site.hs b/builder/site.hs
index c5869dd..a56877b 100644
--- a/builder/site.hs
+++ b/builder/site.hs
@@ -66,7 +66,7 @@ main = mkContext >>= \ctx -> hakyllWith configuration $ do
 
 
 configuration :: Configuration
-configuration = defaultConfiguration{ providerDirectory = "site" }
+configuration = defaultConfiguration{ providerDirectory = "site", previewHost = "0.0.0.0" }
 
 parseTestimonialCompiler :: Compiler (Item Testimonial)
 parseTestimonialCompiler = do
```

## Demo

https://user-images.githubusercontent.com/13085980/216684916-37d97c75-b324-4978-9beb-a2388bd3e8d7.mp4

### CODE REVIEW

**Comment by tomjaguarpaw** [comment]
For me, on desktop, it doesn't seem to allow entering space or numerals.

**Comment by andys8** [comment]
Oh, man, then it's not as easy as applying the suggested fix. Will take a look :D

**Comment by andys8** [comment]
I pushed a change where copied the [file from the fork](https://github.com/chrisdone/jquery-console/blob/85fab01f3fcbf77158993037d93068e0e9d88e86/jquery.console.js) over as is.

The result seems to be that it's working on Android and Desktop. Before I could confirm the same behavior you described.

Why didn't I do this in the first place? The fork has some formatting chances, so I thought I could only use the lines that fix logic. But also: The [file of the fork has 845 lines](https://github.com/chrisdone/jquery-console/blob/85fab01f3fcbf77158993037d93068e0e9d88e86/jquery.console.js) while the file in the [haskell-infra repository has only 750 lines](https://github.com/haskell-infra/www.haskell.org/blob/master/site/js/jquery.console.js). The difference is significant. I haven't looked into it and don't know if these differences are improvements, or code that was actively stripped away to run it on haskell.org.

**Comment by tomjaguarpaw** [comment]
The latest version seems to be fine on desktop.

File: site/js/jquery.console.js [hunk 1]
```diff
@@ -37,714 +37,809 @@
...
+            while (deleteCharAtPos()) {
+                updatePromptDisplay();
+            }
+        };
+
+        function clearCurrentPrompt() {
+            extern.promptText("");
+        };
+
+        function clearScreen() {
```
Comment (andys8) [inline site/js/jquery.console.js:444]: :information_source: This change is coming from this commit: https://github.com/chrisdone/jquery-console/commit/3d405bf1f1613e5c456159adfb6d1b13512b196a

File: site/js/jquery.console.js [hunk 2]
```diff
@@ -37,714 +37,809 @@
...
+                }
+            } else { // Assume it's a DOM node or jQuery object.
+                inner.append(msg);
+            }
+            newPromptBox();
+        };
+
+        ////////////////////////////////////////////////////////////////////////
+        // Report some message into the console
+        function report(msg, className) {
```
Comment (andys8) [inline site/js/jquery.console.js:588]: :information_source: This change is coming from this commit: https://github.com/chrisdone/jquery-console/commit/c8812e4a2a3a4328775391bff06a77aa42753bd0

**Comment by andys8** [comment]
@tomjaguarpaw Good to merge?

**Review by TikhonJelvis — APPROVED** [review]
Makes sense to me, but would be great to have @tomjaguarpaw give it a quick test too.

**Comment by tomjaguarpaw** [comment]
When I tried it on desktop it was fine so let's give this a go. I don't really understand how it works, but if anyone spots any problems after merge we can always roll back.

**Comment by andys8** [comment]
>  if anyone spots any problems after merge we can always roll back

Exactly. Thanks for merging

**Comment by tomjaguarpaw** [comment]
Thanks for contributing!

**Pull request merged**